fondant/dataset.py

```diff
@@ -27,43 +26,73 @@ def __init__(self, manifest: Manifest):
             "__null_dask_index__": "int64",
         }
 
-    def _load_subset(self, name: str, fields: t.List[str]) -> dd.DataFrame:
-        # get subset from the manifest
-        subset = self.manifest.subsets[name]
-        # get remote path
+    def _load_subset(
+        self, subset_name: str, fields: t.List[str], index: t.Optional[Index] = None
+    ) -> dd.DataFrame:
+        """
+        Function that loads the subset
+        Args:
+            subset_name: the name of the subset to load
+            fields: the fields to load from the subset
+            index: optional index to filter the subset on. If not provided, the default manifest
```

Comment:
Shouldn't this always be the manifest index? I don't see this parameter currently being used.